velox/exec/tests/AggregationTest.cpp

```diff
@@ -1245,6 +1245,91 @@ TEST_F(AggregationTest, groupingSets) {
       "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY ROLLUP (k1, k2)");
 }
 
+const std::shared_ptr<const core::GroupIdNode> findGroupIdNode(
+    const core::PlanNodePtr& root,
+    const core::PlanNodeId& groupIdNodeId) {
+  std::vector<core::PlanNodePtr> nodes;
+  nodes.push_back(root);
+  while (!nodes.empty()) {
+    std::vector<core::PlanNodePtr> nextNodes;
+    for (const auto& node : nodes) {
+      if (node->id() == groupIdNodeId) {
+        return std::dynamic_pointer_cast<const core::GroupIdNode>(node);
+      }
+      const auto childNodes = node->sources();
+      std::copy(
+          childNodes.begin(), childNodes.end(), std::back_inserter(nextNodes));
+    }
+    nodes.swap(nextNodes);
+  }
+  VELOX_UNREACHABLE("Plan node {} is not found", groupIdNodeId);
+  return nullptr;
+}
+
+TEST_F(AggregationTest, groupingSetsOutput) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"k1", "k2", "a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 11; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 17; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<StringView>(
+              size,
+              [](auto row) { return StringView(std::string(row % 12, 'x')); }),
+      });
+
+  createDuckDbTable({data});
+
+  core::PlanNodeId randomGroupId;
```

Comment:
Why is it called 'random'? Where does the source of randomness comes from? I'm seeing that the difference between 2 plans is the order of grouping keys in the output. Perhaps, rename to naturalOrder and reversedOrder or similar.